src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ActionListExecutorTests.cs

```diff
@@ -75,23 +92,51 @@ public async Task ActionListExecutor_SecondActionFail()
 
                 CollectionRuleContext context = new(DefaultRuleName, ruleOptions, null, logger);
 
+                bool calledStartCallback = false;
+                Action startCallback = () => calledStartCallback = true;
+
                 CollectionRuleActionExecutionException actionExecutionException = await Assert.ThrowsAsync<CollectionRuleActionExecutionException>(
-                    () => executor.ExecuteActions(context, cancellationTokenSource.Token));
+                    () => executor.ExecuteActions(context, startCallback, cancellationTokenSource.Token));
 
                 Assert.Equal(1, actionExecutionException.ActionIndex);
 
                 Assert.Equal(string.Format(Strings.ErrorMessage_NonzeroExitCode, "1"), actionExecutionException.Message);
+
+                if (waitForCompletion)
```

Comment:
I realize it's really simple, so up to you, but for code reuse should this be put into a method? Just thinking that if we bulk up `ActionListExecutorTests` in the future we're going to have these lines repeated in even more places.